### PR TITLE
S2 Translations

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

In the future, as we bring more strings in from v3, you can add them to the allowed set. This means we can add strings to v3 without worry of polluting s2.

To refresh the locale files, just run `node ./scripts/migrateIntl.mjs`

Strings are added in this format
`packagename.stringKey`
This prevents key collisions across components because localization for a string is dependent on context, so we can't assume all duplicate keys will yield the same translations.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

## 📝 Test Instructions:

Make sure not to just test that the strings are translated, but also that RTL is working.

I know that ColorArea/ColorSlider/ColorWheel aren't quite right yet.
Switch isn't right yet either
tooltip arrow isn't right yet

### CODE REVIEW

File: packages/@react-spectrum/s2/intl/ar-AE.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  "table.sortAscending": "فرز بترتيب تصاعدي",
+  "table.sortDescending": "فرز بترتيب تنازلي",
+  "tag.actions": "الإجراءات",
+  "tag.hideButtonLabel": "إظهار أقل",
+  "tag.noTags": "بدون",
+  "tag.showAllButtonLabel": "عرض الكل ({tagCount, number})",
+  "toast.info": "معلومات",
+  "toast.negative": "خطأ",
+  "toast.positive": "تم بنجاح"
+}
```
[the remark this answers is not in the document]
Comment (snowystinger) [reply packages/@react-spectrum/s2/intl/ar-AE.json:72]: yeah, i have a note where i'm tracking the unused ones for removal
https://github.com/adobe/react-spectrum/pull/6878/files#diff-e7c346dd76ce8b3632a6e9d77ffc549ebff92bb608478a9b9241121f45095a51R77

**Comment by snowystinger** [comment]
Found there is a way to avoid having our own addon for the locale switching, however, it turned out to be janky on the "docs" pages when changing the locale and would refresh the entire page.
I've added a commit for posterity https://github.com/adobe/react-spectrum/pull/6878/commits/03be283ff3b13c0ab4d7e0dd88d4cd069e606c27

**Review by LFDanLu — DISMISSED** [review]
Verified that the strings are being applied properly to the components in various locales. Provisional approval pending any changes/callouts needed for the localization team so they don't retranslate these strings

**Review by devongovett — APPROVED** [review]
Updated build script to skip S2 strings in v3 monopackage

**Review by LFDanLu — APPROVED** [review]

**Pull request merged**